cclib/bridge/cclib2pyquante.py

```diff
@@ -1,31 +1,72 @@
 # -*- coding: utf-8 -*-
 #
-# Copyright (c) 2017, the cclib development team
+# Copyright (c) 2020, the cclib development team
 #
 # This file is part of cclib (http://cclib.github.io) and is distributed under
 # the terms of the BSD 3-Clause License.
 
 """Bridge for using cclib data in PyQuante (http://pyquante.sourceforge.net)."""
 
+import numpy
 from cclib.parser.utils import find_package
 
+
+class MissingAttributeError(Exception):
+    pass
+
+
+# Support for old PyQuante may be dropped in future release.
 _found_pyquante = find_package("PyQuante")
+_found_pyquante2 = find_package("pyquante2")
 if _found_pyquante:
     from PyQuante.Molecule import Molecule
+if _found_pyquante2:
+    from pyquante2 import molecule
 
 
-def _check_pyquante(found_pyquante):
-    if not found_pyquante:
+def _check_pyquante():
+    if not _found_pyquante and not _found_pyquante2:
         raise ImportError("You must install `PyQuante` to use this function")
 
 
-def makepyquante(atomcoords, atomnos, charge=0, mult=1):
-    """Create a PyQuante Molecule."""
-    _check_pyquante(_found_pyquante)
-    return Molecule("notitle",
-                    list(zip(atomnos, atomcoords)),
-                    units="Angstrom",
-                    charge=charge, multiplicity=mult)
+def makepyquante(data):
+    """Create a PyQuante Molecule from ccData object."""
+    _check_pyquante()
+
+    # Check required attributes.
+    required_attrs = {"atomcoords", "atomnos"}
+    missing = [x for x in required_attrs if not hasattr(data, x)]
+
+    if missing:
+        missing = " ".join(missing)
+        raise MissingAttributeError(
+            "Could not create pyquante molecule due to missing attribute: " + missing

```

Comment:
`... attribute: {}".format(missing)` is a little neater

Reply:
I've updated just updated this. Thanks for the suggestion!